src/controllers/dynakube/version/oneagent_test.go

```diff
@@ -0,0 +1,115 @@
+package version
+
+import (
+	"testing"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/dockerconfig"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects/address"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOneAgentUpdater(t *testing.T) {
+	testImage := dtclient.LatestImageInfo{
+		Source: "some.registry.com",
+		Tag:    "1.2.3",
+	}
+	t.Run("Getters work as expected", func(t *testing.T) {
+		dynakube := &dynatracev1beta1.DynaKube{
+			Spec: dynatracev1beta1.DynaKubeSpec{
+				OneAgent: dynatracev1beta1.OneAgentSpec{
+					ClassicFullStack: &dynatracev1beta1.HostInjectSpec{
+						AutoUpdate: address.Of(false),
+						Image:      testImage.Uri(),
+						Version:    testImage.Tag,
+					},
+				},
+			},
+		}
+		registry := newEmptyFakeRegistry()
+		mockClient := &dtclient.MockDynatraceClient{}
+		mockOneAgentImageInfo(mockClient, testImage)
+		updater := newOneAgentUpdater(dynakube, mockClient, registry.ImageVersionExt)
+
+		assert.Equal(t, "oneagent", updater.Name())
+		assert.True(t, updater.Enabled())
+		assert.Equal(t, dynakube.Spec.OneAgent.ClassicFullStack.Image, updater.CustomImage())
+		assert.Equal(t, dynakube.Spec.OneAgent.ClassicFullStack.Version, updater.CustomVersion())
+		assert.False(t, updater.AutoUpdate())
+		imageInfo, err := updater.LatestImageInfo()
+		require.NoError(t, err)
+		assert.Equal(t, testImage, *imageInfo)
+	})
+}
+
+func TestOneAgentUseDefault(t *testing.T) {
```

Comment:
Tabletests are not that great for me as VSCode does not let me run them individually AND IMO they are hell to debug. (for the same reason)

but I agree I could abstract parts of the tests 😅

Reply:
Get a better IDE.

Reply:
the IDE issue is the smaller one, the debuging part is the big one.

if you have 10 test cases, 2 of which fails.
- in case of a table test the trace will point at the same line, so finding out which case failed is annoying to find out
- in the current case the trace points at the exact test case that failed (and I can actually click on said trace, that will open the IDE there)

furthermore, if you correctly abstract the common parts of your tests (which I didn't do yet), then the outcome is the same or less code then in case of a table test. (and it is also less brittle)